shared-libs/rules-engine/src/provider-wireup.js

```diff
@@ -135,6 +139,48 @@ module.exports = {
   },
 };
 
+let refreshQueue = Promise.resolve();
+const enqueue = callback => {
+  const listeners = [];
+  const eventQueue = [];
+  const emit = evtName => {
+    if (!listeners[evtName]) {
+      return eventQueue.push(evtName);
+    }
+    listeners[evtName].forEach(callback => callback());
+  };
+
+  emit('queued');
+  refreshQueue = refreshQueue.then(() => {
+    emit('running');
+    return callback();
+  });
+
+  refreshQueue.on = (evtName, callback) => {
+    listeners[evtName] = listeners[evtName] || [];
+    listeners[evtName].push(callback);
+    eventQueue.forEach(queuedEvent => queuedEvent === evtName && callback());
+    return refreshQueue;
+  };
+
+  return refreshQueue;
+};
+
+const jumpQueue = result => {
+  const listeners = [];
+  const emit = evtName => (listeners[evtName] || []).forEach(callback => callback());
+  const p = Promise.resolve().then(() => {
+    emit('running');
```

Comment:
Is it even worth emitting this? I mean, it's _not_ running, right? Why store task telemetry for users who have tasks disabled?

Reply:
Perfect. 
I initially wrote this really simplistically like p.on(() => p) but then changed my mind. I'm going to revert to the simple version.